components/filter-header/filter-header.js

```diff
@@ -1,25 +1,81 @@
-class FilterHeader extends crsbinding.classes.BindableElement {
+/**
+ * @class FilterHeader - It's a custom element that filters the children of a container element.
+ *
+ * Features:
+ * filter - filters the children of the container element
+ * close - closes the filter
+ *
+ */
+class FilterHeader extends HTMLElement {
     #container;
-
-    get shadowDom() {
-        return true;
-    }
+    #filterHandler = this.filter.bind(this);
 
     get html() {
         return import.meta.url.replace(".js", ".html");
     }
 
+    constructor() {
+        super();
+        this.attachShadow({ mode: "open" });
+    }
+
+    /**
+     * @method connectedCallback - When the component is added to the DOM, the container element is set to the element
+     * that has the id that is specified in the "for" attribute.
+     */
     async connectedCallback() {
-        super.connectedCallback();
-        const query = this.getAttribute("for");
-        this.#container = this.parentElement.querySelector(query);
+        this.shadowRoot.innerHTML = await fetch(this.html).then(result => result.text());
+        await this.load();
+    }
+
+    /**
+     * @method load - It waits for the DOM to be ready, then it adds an event listener to the input element, and then it notifies the
+     * framework that the component is ready
+     * @returns A promise that resolves when the component is ready.
+     */
+    load() {
+        return new Promise(resolve => {
+            requestAnimationFrame(async () => {
+                const query = this.getAttribute("for");
+                this.#container = this.parentElement.querySelector(query);
+                this.shadowRoot.querySelector("input").addEventListener("keyup", this.#filterHandler);
+                await crs.call("component", "notify_ready", {
+                    element: this
+                });
+
+                const placeholder = {
+                    placeholder: "Type here to filter"
+                }
+
+                await crsbinding.translations.add(placeholder, "filterHeader");
+                await crsbinding.translations.parseElement(this.shadowRoot.querySelector("input"));
+                resolve();
+            });
+        });
     }
 
+
+    /**
+     * @method disconnectedCallback - When the component is removed from the DOM, the container is set to null
+     */
     async disconnectedCallback() {
         this.#container = null;
-        await super.disconnectedCallback();
+        this.shadowRoot.querySelector("input").removeEventListener("keyup", this.#filterHandler);
+        await crsbinding.translations.delete("filterHeader");
+        this.#filterHandler = null;
```

Comment:
check the memory tool

Reply:
Checked with tool, no memory leaks were found